gulp/tasks/templates.js

```diff
@@ -1,55 +1,97 @@
-var _ = require('lodash'); // to handle collections gracefully
-var config = require('../config').templates; // pull in the pathing config file
-var fs = require('fs'); // used to work with substack's module
-var glob = require('glob'); // to dynamically read in all content md files
-var gulp = require('gulp'); // because this is a gulp task. duh.
-var HTMLtemplate = require('html-template'); // substack's html template implementation
-var md = require('markdown-it')({ html: true }); // to convert markdown to html
-var source = require('vinyl-source-stream'); // used to convert substack's readStream to vinylStream
+var _             = require('lodash'); // to handle collections gracefully
+var config        = require('../config').templates; // pull in the pathing config file
+var versions      = require('../config').versions; // pull in the pathing config file
+var fs            = require('fs'); // used to work with substack's module
+var glob          = require('glob'); // to dynamically read in all content md files
+var gulp          = require('gulp'); // because this is a gulp task. duh.
+var HTMLtemplate  = require('html-template'); // substack's html template implementation
+var md            = require('markdown-it')({ html: true }); // to convert markdown to html
+var source        = require('vinyl-source-stream'); // used to convert substack's readStream to vinylStream
 var replaceStream = require('replacestream'); // used to add an element to the html: making sure each page has it's own class if needed in css
-var moment = require('moment-timezone');
-var exec = require('child_process').exec
-var utils = require('../util/template-utils.js');
-var path = require('path');
+var moment        = require('moment-timezone');
+var exec          = require('child_process').exec
+var utils         = require('../util/template-utils.js');
+var path          = require('path');
+var injectVersions;
+
+
+//  function to inject the versions into the string given
+injectVersions = function(string, versions){
+
+  if( !Array.isArray(versions) ){
+    throw new Error('expecting an array of versions!');
+  }
+
+  versions
+  .forEach(function(thisVersion){
+
+    if(thisVersion.linkRegExp !== 'undefined'){
```

Comment:
I think you either meant `undefined` instead of `'undefined'`, or are missing an `typeof` in these.